ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/TransactionComponents.kt

```diff
@@ -0,0 +1,196 @@
+    fun openAsJar(): JarInputStream {
+        val stream = open()
+        return try {
+            JarInputStream(stream)
+        } catch (ex: IOException) {
+            stream.use { throw ex }
+        }
+    }
+}
+
+/**
+ * Defines an interval on a timeline; not a single, instantaneous point.
+ *
+ * There is no such thing as "exact" time in distributed systems, due to the underlying physics involved, and other
+ * issues such as network latency. A time window represents an approximation of an instant with a margin of tolerance,
+ * and may be fully bounded, or unbounded towards negative or positive infinity.
+ *
+ * @property from The boundary at which the time window begins, or null if the time window is unbounded towards negative infinity.
+ * @property until The boundary at which the time window ends, or null if the time window is unbounded towards positive infinity.
+ * @property midpoint The midpoint between a fully bounded time window, or null if the time window is unbounded towards positive or negative infinity.
+ * @property duration The duration of a fully bounded time window, or null if the time window is unbounded towards positive or negative infinity.
+ */
+@DoNotImplement
+@CordaSerializable
+interface TimeWindow {
+    val from: Instant?
+    val until: Instant?
+    val midpoint: Instant?
+    val duration: Duration? get() = if (from == null || until == null) null else Duration.between(from, until)
+
+    /**
+     * Determines whether the current [TimeWindow] contains the specified [Instant].
+     *
+     * @param instant The [Instant] to check is contained within the current [TimeWindow].
+     * @return Returns true if the current [TimeWindow] contains the specified [Instant]; otherwise, false.
+     */
+    operator fun contains(instant: Instant): Boolean
+}
+
+/**
+ * Defines information that applies to a transaction state.
+ *
+ * @property contractId The class name of the [Contract] associated with the transaction state.
+ * @property notary The [Party] of the notary associated with the transaction state.
+ * @property encumbrance The index of an associated, encumbered state, or null if no encumbrance applies to the associated transaction state.
+ * @property constraint The [CpkConstraint] associated with the transaction state.
+ */
+@DoNotImplement
+@CordaSerializable
+interface TransactionStateInformation {
+    val contractId: String
+    val notary: Party
+    val encumbrance: Int?
+    val constraint: CpkConstraint
+}
+
+/**
+ * Defines a transaction state, composed of a [ContractState] and associated [TransactionStateInformation].
+ *
+ * @param T The underlying type of the [ContractState] instance.
+ * @property contractState The [ContractState] component of the current [TransactionState] instance.
+ * @property information The [TransactionStateInformation] component of the current [TransactionState] instance.
+ */
+@DoNotImplement
+@CordaSerializable
+interface TransactionState<out T : ContractState> {
```

Comment:
Similar problem to the command signers in that this will need to be split up under the hood to allow partial evidence without sending/deserializing the states. Maybe we allow a synthesized view, but maybe we do just slit this stuff up in some way on the UtxoLedgerTransaction?